lib/taxjar/api/request.rb

```diff
@@ -72,17 +72,16 @@ def symbolize_keys!(object)
           object
         end
 
-        def fail_or_return_response_body(code, body)
-          e = extract_error(code, body)
-          fail(e) if e
-          body[object_key.to_sym]
-        end
-
-        def extract_error(code, body)
-          return Taxjar::Error.for_json_parse_error(code) if body.nil?
-          klass = Taxjar::Error::ERRORS[code]
-          if !klass.nil?
-            klass.from_response(body)
+        def fail_or_return_response_body(response, body)
+          if body.nil?
+            fail(Taxjar::Error.for_json_parse_error(response.code))
+          elsif response.status.success?
+            body[object_key.to_sym]
+          elsif !(klass = Taxjar::Error::ERRORS[response.code]).nil?
+            fail(klass.from_response(body))
+          else
+            message = HTTP::Response::Status::REASONS[response.code] || "Unknown Error"
```

Comment:
@poops If a developer rescues an `Unknown Error` exception, can we return back the response code as well? We can assume a class name (`Taxjar::Error`), message, and response code may want to be examined.

Reply:
Actually return a code either way for a generic `Taxjar::Error` in this `else` block.